rust/lance-index/src/vector/hnsw.rs

```diff
@@ -70,15 +57,18 @@ pub fn select_neighbors_heuristic(
     candidates.sort_unstable_by(|a, b| b.dist.partial_cmp(&a.dist).unwrap());
 
     let mut results: Vec<OrderedNode> = Vec::with_capacity(k);
-    while !candidates.is_empty() && results.len() < k {
-        let u = candidates.pop().unwrap();
+    for u in candidates.iter().rev() {
```

Comment:
I think the .rev here is probably your bug

Reply:
Could you elaborate?